features/config/service/api/src/main/java/org/opennms/features/config/service/api/ConfigurationManagerService.java

```diff
@@ -0,0 +1,132 @@
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.config.service.api;
+
+import org.json.JSONObject;
+import org.opennms.features.config.dao.api.ConfigConverter;
+import org.opennms.features.config.dao.api.ConfigData;
+import org.opennms.features.config.dao.api.ConfigSchema;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Responsible for managing Schemas and Configurations.
+ * A Schema is a set of rules that constrain the data of a Configuration.
+ * A Configuration is the data that defines the runtime behaviour of a service together with the code of that service.
+ */
+public interface ConfigurationManagerService {
+
+    /**
+     * Register a service to use config manager
+     *
+     * @param configName
+     * @param majorVersion
+     * @param minorVersion
+     * @param patchVersion
+     * @param entityClass (Must have ValidateUsing annotation)
+     * @param <ENTITY>
+     * @throws IOException
+     * @throws ClassNotFoundException
+     */
+    <ENTITY> void registerSchema(final String configName, final int majorVersion, final int minorVersion,
+                                 final int patchVersion, Class<ENTITY> entityClass) throws IOException, ClassNotFoundException;
+
+    void registerSchema(final String configName, final int majorVersion, final int minorVersion,
+                        final int patchVersion, final ConfigConverter converter) throws IOException, ClassNotFoundException;
+
+    /**
+     * Get the registered Schema
+     *
+     * @param configName
+     * @return ConfigSchema
+     * @throws IOException
+     */
+    Optional<ConfigSchema<?>> getRegisteredSchema(String configName) throws IOException, ClassNotFoundException;
+
+    /**
+     * register a new configuration by xml
+     *
+     * @param configName
+     * @param configId
+     * @param xmlPath
+     * @throws IOException
+     */
+    void registerConfiguration(String configName, String configId, String xmlPath) throws IOException, ClassNotFoundException;
```

Comment:
I am not sure if we need this method since we don't want to read xml from the file system anymore. For the existing configurations we could do the migration via Liquibase (and a helper class there that reads xml from the file system)

Reply:
changed to configName

Reply:
Ok removed.